Enchanted/UI/macOS/Menus.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  Menus.swift
+//  Enchanted
+//
+//  Created by Wildan Zulfikar on 24.4.2024.
+//
+
+import Foundation
+import SwiftUI
+
+#if os(macOS)
+struct ShowSettingsKey: FocusedValueKey {
+    typealias Value = Binding<Bool>
+}
+
+extension FocusedValues {
+    var showSettings: Binding<Bool>? {
+        get { self[ShowSettingsKey.self] }
+        set { self[ShowSettingsKey.self] = newValue }
+    }
+}
```

Comment:
Not sure if this is the best place to put these code, happy to move if you have feedback

Reply:
Makes sense to me.